### PR TITLE
Blog: Publishing Modbus data to your UNS with FlowFuse

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/11/bridging-modbus-to-mqtt.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
+---
+title: Bridging Modbus Data to MQTT using FlowFuse
+subtitle: Seamlessly Integrating Modbus with MQTT for Smart Industrial Automation using FlowFuse
+description: Learn how to bridge Modbus data to MQTT using Node-RED for cloud integration and real-time industrial monitoring.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/bridging-modbus-to-mqtt.md:4]: Need to mention FlowFuse

File: src/blog/2024/11/bridging-modbus-to-mqtt.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+image: /blog/2024/11/images/bridging-modbus-to-mqtt.png
+keywords: modbus to mqtt, mqtt to modbus, node-red as gateway, bridging modbus to mqtt
+tags:
+   - node-red
+   - mqtt
+   - opcua
+   - modbus
+---
+
+Modbus is a popular protocol in industrial automation used to connect devices like sensors and PLCs, but it’s not designed to connect easily with cloud platforms or modern IoT systems. MQTT, however, is well-suited for real-time data transfer to the cloud. Using Node-RED, you can bridge the gap between Modbus and MQTT, allowing older equipment to connect to modern IoT networks. Let’s take a look at how you can set this up.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/bridging-modbus-to-mqtt.md:16]:
```suggestion
Modbus is a popular protocol in industrial automation used to connect devices like sensors and PLCs, but it’s not designed to connect easily with cloud platforms or modern IoT systems. MQTT, however, is well-suited for real-time data transfer to the cloud. Using Node-Red, you can bridge the gap between Modbus and MQTT, allowing older equipment to connect to modern IoT networks. Let’s take a look at how you can set this up on the FlowFuse Platform. 
```

File: src/blog/2024/11/bridging-modbus-to-mqtt.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+Once you’ve configured the MQTT nodes for all your metrics and deployed the flow, check the status at the bottom of each MQTT node. If it shows "connected," your Node-RED flow is successfully connected to the broker and publishes data to the MQTT broker. From here, you can integrate the data with cloud-based analytics platforms. Build IoT dashboards with [FlowFuse Dashboard](https://dashboard.flowfuse.com/), or other systems to enable real-time monitoring, predictive maintenance, and automated decision-making. This setup effectively bridges the gap between legacy Modbus devices and modern IoT infrastructure, empowering smarter, more efficient industrial operations.
+
+To monitor the data being sent to the Broker, you can use MQTT client monitoring software, such as MQTT Explorer, or similar tools.
+
+![Image showing the MQTT Explorer interface monitoring an MQTT broker](./images/mqtt-client-explorer.png){data-zoomable}
+_Image showing the MQTT Explorer interface monitoring an MQTT broker_
+
+## Final Thought
+
```
Comment (Hasmin-AC) [inline src/blog/2024/11/bridging-modbus-to-mqtt.md:245]: There's no mention of FlowFuse at all in this conclusion. I would suggest editing this part accordingly.

File: src/blog/2024/11/bridging-modbus-to-mqtt.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
+---
+title: Bridging Modbus Data to MQTT using FlowFuse
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/bridging-modbus-to-mqtt.md:2]: Alternatively we could make this UNS focussed, what do you think? `Publishing Modbus data to your UNS with FlowFuse`?

Comment (ZJvandeWeg) [reply src/blog/2024/11/bridging-modbus-to-mqtt.md:2]: let's also consider updating the filename and thus the blog URL path

**Comment by ZJvandeWeg** [comment]
@sumitshinde-84 Still some unresolved notes, is this ready for review?

File: src/blog/2024/11/publishing-modbus-data-to-uns.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+
+For instance, in a factory still relying on Modbus, machines such as motors, pumps, and conveyors can send their operational data to the cloud via MQTT. Once the data is in the cloud, it can be published into the Unified Namespace (UNS), giving stakeholders easy access to a centralized view of all device data in real time. This integration provides immediate insights into machine performance, enabling operators to track temperature, vibration, and runtime metrics. With these insights, predictive maintenance models can be applied, helping to identify potential failures before they occur and preventing unplanned downtime.
+
+## How to Bridge Modbus to MQTT
+
+This section will cover the steps to bridge Modbus data to MQTT using FlowFuse, leveraging Node-RED's capabilities. The process involves retrieving data from a Modbus device (For Practical example, we are using OpenSim to simulate Modbus data), transforming and processing the data (e.g., scaling raw sensor data into human-readable formats), and sending it to your Unified Namespace.
```
Comment (Hasmin-AC) [inline src/blog/2024/11/publishing-modbus-data-to-uns.md:35]:
```suggestion
Let's look at the steps to bridge Modbus data to MQTT using FlowFuse, leveraging Node-RED's capabilities. The process involves retrieving data from a Modbus device (For Practical example, we are using OpenSim to simulate Modbus data), transforming and processing the data (e.g., scaling raw sensor data into human-readable formats), and sending it to your Unified Namespace.
```

**Review by Hasmin-AC — APPROVED** [review]

File: src/blog/2024/11/images/mqtt-client-explorer.png
Comment (ZJvandeWeg) [inline src/blog/2024/11/images/mqtt-client-explorer.png:1]: Can we use the FF topic browser?

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

File: src/blog/2024/11/publishing-modbus-data-to-uns.md [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+
+<!--more-->
+
+## The Importance of Bridging Modbus Data to MQTT for Cloud Integration and UNS
+
+Modbus has long been a foundational protocol in industrial environments, enabling reliable communication between devices like PLCs, sensors, and motors. It has served the needs of operational technology (OT) for decades. However, as industries embrace Industry 4.0, there is an increasing need to bridge the gap between legacy OT systems like Modbus and modern information technology (IT) platforms, including cloud-based systems and IoT applications. This is where the challenge of the OT/IT gap arises.
+
+While Modbus excels in local control and data collection, it struggles to integrate with cloud-based platforms and real-time IoT solutions. That's where MQTT, a lightweight, real-time messaging protocol, becomes essential. MQTT allows devices to transmit data efficiently to the cloud, even in environments with limited bandwidth, power, or connectivity. However, bridging them is crucial since Modbus and MQTT don’t natively communicate for bridging. We will need to use a platform like Flowfuse that supports both.
```
Comment (ZJvandeWeg) [inline src/blog/2024/11/publishing-modbus-data-to-uns.md:26]: This paragraph needs to be about UNS, not MQTT. MQTT is a broker _choice_ for the UNS which is the data architecture to communicate with arbitrary consumers. For our UNS we use MQTT, which then links to your other post about setting up a UNS in 15 minutes?

**Review by ZJvandeWeg — CHANGES_REQUESTED** [review]

**Review by Hasmin-AC — APPROVED** [review]

File: src/blog/2024/12/publishing-modbus-data-to-uns.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
+---
+title: Publishing Modbus data to your UNS with FlowFuse
+subtitle: Bridging Modbus Data to MQTT using FlowFuse
+description: Learn how to bridge Modbus data to MQTT and publish it to a Unified Namespace (UNS) using FlowFuse for real-time monitoring and cloud integration.
+date: 2024-12-03
```
Comment (ZJvandeWeg) [inline src/blog/2024/12/publishing-modbus-data-to-uns.md:5]:
```suggestion
date: 2024-12-04
```

**Review by ZJvandeWeg — APPROVED** [review]

**Pull request merged**